src/components/postgresConnect.ts

```diff
@@ -648,10 +648,19 @@ class PostgresDB {
 
   public joinPermissionNameToUserOrg = async (
     username: string,
-    orgName: string,
+    org: string | number,
     permissionName: string
   ) => {
-    const text = `
+    const text =
```

Comment:
How about just:
```
const text =
    INSERT INTO permission_join (user_id, organisation_id, permission_name_id) 
    values (
        (select id from "user" where username = $1),
        ${typeof org === 'number' ? "$2" : "(select id from organisation where name = $2)"},
        (select id from permission_name where name = $3))
    ON CONFLICT (user_id, organisation_id, permission_name_id)
      WHERE organisation_id IS NOT NULL
    DO
    		UPDATE SET user_id = (select id from "user" where username = $1)
    RETURNING id
    `
```
I think the `ON CONFLICT...  DO....` stuff is important too, and you didn't have it in your query, was that intentional?

Reply:
Good point, updating to this now